lib/collection.js

```diff
@@ -1515,6 +1515,13 @@ Collection.prototype.findOneAndReplace = function(filter, replacement, options,
   if (replacement == null || typeof replacement !== 'object')
     throw toError('replacement parameter must be an object');
 
+  // Check that there are no atomic operators
+  const keys = Object.keys(replacement);
+
+  if (keys[0] && keys[0][0] === '$') {
```

Comment:
nit: I think we need: `if (keys[0] && Array.isArray(keys[0]) && keys[0][0]...`, also a test would be nice to prove that we do this extra validation

Reply:
we're not looking for an array here, we're expecting `keys[0]` to be a string. We're checking the first character. Anything that has `keys[0]` as truthy will not throw on `keys[0][0]`, so we should be fine.

Reply:
:+1: